### PR TITLE
Update lobby services

### LINKED ISSUE #151: Implement the services to interact with the lobby: update lobby

### PR DESCRIPTION
**Pull request by paolovic**
closes #151

### CODE REVIEW

File: src/main/java/ch/uzh/ifi/hase/soprafs22/service/LobbyService.java [hunk 1]
```diff
@@ -35,49 +36,54 @@
...
 
     @Autowired
-    public LobbyService(UserRepository userRepository){
+    public LobbyService(UserRepository userRepository) {
         this.userRepository = userRepository;
         this.lobbyManager = LobbyManager.getInstance();
     }
 
+    private void checkLobbyNameNullOrEmpty(String lobbyName, String errorMessageEnding) {
+        if (lobbyName == null || lobbyName.isEmpty()) {
```
Comment (hilalcomak) [inline src/main/java/ch/uzh/ifi/hase/soprafs22/service/LobbyService.java:45]: Does this mean that now "     " is a valid lobby name?

File: src/main/java/ch/uzh/ifi/hase/soprafs22/service/LobbyService.java [hunk 2]
```diff
@@ -141,4 +149,34 @@ public ILobby getLobby(String token, Long lobbyId){
...
+    public void updateLobby(ILobby lobby, String token, String lobbyName, Visibility visibility, GameMode gameMode, GameType gameType) {
+        if (!token.equals(lobby.getHost().getToken())) {
+            throw new ResponseStatusException(HttpStatus.UNAUTHORIZED, "User is not the host of the lobby.");
+        }
+
+        // Check if values are valid
+        checkLobbyNameNullOrEmpty(lobbyName, "updated");
+        checkEnumConfigNull(visibility, "visibility", "updated");
+        checkEnumConfigNull(gameMode, "game mode", "updated");
+        checkEnumConfigNull(gameType, "game type", "updated");
```
Comment (hilalcomak) [inline src/main/java/ch/uzh/ifi/hase/soprafs22/service/LobbyService.java:161]: Does this mean that the client now has to provide everything (visibility, game mode, etc.) to update the lobby name?

File: src/test/java/ch/uzh/ifi/hase/soprafs22/service/Integration/LobbyServiceIntegrationTests.java [hunk 3]
```diff
@@ -211,5 +212,92 @@ void getLobby_wrongToken_throwException() throws SmallestIdNotCreatableException
...
+        createdLobby.setGameMode(gameMode);
+        createdLobby.setGameType(gameType);
+        Long id = createdLobby.getId();
+
+        // Attempt to update lobby with empty name
+        ResponseStatusException exception = Assertions.assertThrows(ResponseStatusException.class,
+                () -> lobbyService.updateLobby(createdLobby, createdLobby.getHost().getToken(), "", visibility, gameMode, gameType));
 
+        // Check https status code
+        assertEquals(exception.getRawStatusCode(), HttpStatus.BAD_REQUEST.value());
```
Comment (hilalcomak) [inline src/test/java/ch/uzh/ifi/hase/soprafs22/service/Integration/LobbyServiceIntegrationTests.java:257]: assertEquals(expected, actual)
assertEquals(HttpStatus.BAD_REQUEST, exception.getStatus());

File: src/test/java/ch/uzh/ifi/hase/soprafs22/lobby/LobbyManagerUpdateTest.java [hunk 4]
```diff
@@ -34,114 +35,4 @@ public void before() throws SmallestIdNotCreatableException {
         LOBBY1 = lobbyManager.createLobby("Lobby 1", Visibility.PUBLIC);
         //lobbyManager.addLobby(LOBBY2);
     }
-
```
Comment (hilalcomak) [inline src/test/java/ch/uzh/ifi/hase/soprafs22/lobby/LobbyManagerUpdateTest.java:37]: Please remove file, no point keeping a *Test file with no tests in it.

**Review by hilalcomak — APPROVED** [review]

File: src/main/java/ch/uzh/ifi/hase/soprafs22/service/LobbyService.java [hunk 5]
```diff
@@ -141,4 +149,34 @@ public ILobby getLobby(String token, Long lobbyId){
         throw new ResponseStatusException(HttpStatus.FORBIDDEN, errorMessage);
     }
 
+    public void updateLobby(ILobby lobby, String token, String lobbyName, Visibility visibility, GameMode gameMode, GameType gameType) {
+        if (!token.equals(lobby.getHost().getToken())) {
+            throw new ResponseStatusException(HttpStatus.UNAUTHORIZED, "User is not the host of the lobby.");
```
Comment (apple00juice) [inline src/main/java/ch/uzh/ifi/hase/soprafs22/service/LobbyService.java:154]: I would distinguish between (token == null || token.empty()) and (!token.equals(lobby.getHost().getToken()).
Also UNAUTHORIZED means "unauthenticated" eg. if the token is null or empty. FORBIDDEN is if the token is set but does not match.

File: src/main/java/ch/uzh/ifi/hase/soprafs22/lobby/Lobby.java [hunk 6]
```diff
@@ -19,7 +19,7 @@
     private String name;
     private Visibility visibility;
     private Game game;
-    private final Player owner;
+    private final Player host;
```
Comment (apple00juice) [inline src/main/java/ch/uzh/ifi/hase/soprafs22/lobby/Lobby.java:22]: Didn't we switch to owner to match the json data sent by the endpoint?

**Review by apple00juice — COMMENTED** [review]
Rest looks good, except some of hilals comments, especially that the user has to send everything to the server.
I don't know what happens if a field is not set in the call body, is it then null or does the request fail? If it is just null,  then there could be a null check and just update what was given

**Comment by paolovic** [comment]
> Rest looks good, except some of hilals comments, especially that the user has to send everything to the server. I don't know what happens if a field is not set in the call body, is it then null or does the request fail? If it is just null, then there could be a null check and just update what was given

yes, all fields must be set, if they are null it will throw an exception.

**Comment by paolovic** [comment]
Please, review again @hilalcomak @apple00juice

File: src/main/java/ch/uzh/ifi/hase/soprafs22/service/LobbyService.java [hunk 7]
```diff
@@ -41,10 +40,16 @@ public LobbyService(UserRepository userRepository) {
         this.lobbyManager = LobbyManager.getInstance();
     }
 
-    private void checkLobbyNameNullOrEmpty(String lobbyName, String errorMessageEnding) {
-        if (lobbyName == null || lobbyName.isEmpty()) {
-            String errorMessage = "The lobby name provided is empty. Therefore, the lobby could not be " + errorMessageEnding + "!";
-            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, errorMessage);
+    private void checkStringConfigNullOrEmpty(String s, String errorMessageEnding, boolean isToken) {
```
Comment (apple00juice) [inline src/main/java/ch/uzh/ifi/hase/soprafs22/service/LobbyService.java:43]: I would do two methods one for token and one for the other things.
Also now the error message is always "The lobby name provided....", I would add a parameter for the whole error message.

Comment (apple00juice) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/service/LobbyService.java:43]: Or do it even more general and allow to set the httpstatus too

**Review by apple00juice — APPROVED** [review]
the rest looks good

**Review by paolovic — COMMENTED** [review]
all my comments were invisible for you I assume since they were all pending.
sorry I answered to all of them...

**Pull request merged**